akka-testkit/src/test/java/akka/testkit/javadsl/TestKitApiTest.java

```diff
@@ -0,0 +1,173 @@
+
+public class TestKitApiTest {
+
+  public void compileOnlyTestCase() {
+    TestKit testKit = null;
+    ActorRef actorRef = null;
+    TestActor.AutoPilot autoPilot = null;
+
+    // just coverage of calling all testkit methods, actual impl is tested elsewhere
+    testKit.awaitAssert(
+        () -> {
+          return null;
+        });
+    testKit.awaitAssert(
+        Duration.ofSeconds(3),
+        () -> {
+          return null;
+        });
+    testKit.awaitAssert(
+        Duration.ofSeconds(3),
+        Duration.ofMillis(200),
+        () -> {
+          return null;
+        });
+
+    testKit.awaitCond(
+        () -> {
+          return true;
+        });
+    testKit.awaitCond(
+        Duration.ofSeconds(3),
+        () -> {
+          return true;
+        });
+    testKit.awaitCond(
+        Duration.ofSeconds(3),
+        Duration.ofMillis(200),
+        () -> {
+          return true;
+        });
+    testKit.awaitCond(
+        Duration.ofSeconds(3),
+        Duration.ofMillis(200),
+        "details",
+        () -> {
+          return true;
+        });
+
+    testKit.childActorOf(Props.empty());
+    testKit.childActorOf(Props.empty(), "name");
+    testKit.childActorOf(Props.empty(), SupervisorStrategy.defaultStrategy());
+    testKit.childActorOf(Props.empty(), "name", SupervisorStrategy.defaultStrategy());
+
+    testKit.dilated(Duration.ofSeconds(3));
+
+    testKit.expectMsg("message");
+    testKit.expectMsg(Duration.ofSeconds(3), "message");
+    testKit.expectMsg(Duration.ofSeconds(3), "message", "hint");
+
+    testKit.expectMsgClass(String.class);
+    testKit.expectMsgClass(Duration.ofSeconds(3), String.class);
+
+    testKit.expectMsgEquals("message");
+    testKit.expectMsgEquals(Duration.ofSeconds(3), "message");
+
+    // FIXME why are these called PF when they take a total function, how are they supposed to be
+    // used? #27562
+    testKit.expectMsgPF(
+        Duration.ofSeconds(3),
+        "hint",
+        (value) -> {
+          return null;
+        });
+    testKit.expectMsgPF(
+        "hint",
+        (value) -> {
+          return null;
+        });
+
+    testKit.expectNoMessage();
+    testKit.expectNoMessage(Duration.ofSeconds(3));
+
+    testKit.expectTerminated(actorRef);
+    testKit.expectTerminated(Duration.ofSeconds(3), actorRef);
+
+    // FIXME how is this supposed to be used, scaladoc talks about a partial function but it accepts
+    // a total function #27562
+    Object fishResult1 =
+        testKit.fishForMessage(
+            Duration.ofSeconds(3),
+            "hint",
+            (message) -> {
+              return "fishResult1";
+            });
+    // FIXME how is this supposed to be used, scaladoc talks about a partial function but it accepts
+    // a total function #27562
+    String result =
+        testKit.fishForSpecificMessage(
+            Duration.ofSeconds(3),
+            "hint",
+            (message) -> {
+              return "fishResult2";
```

Comment:
this API makes no sense, how would you say "not matching"

Reply:
I agree, there's a bunch more of them that already existed though. I created #27562 to deal with it but I guess I can remove them in this PR just as well.

Reply:
For clarification: Went for separate issue because I thought it would be good to also provide something that actually works but didn't want to put in that work right now.